src/components/AddBareMetalHosts/OCMClusterDetailTabContent.tsx

```diff
@@ -1,25 +1,50 @@
 import React from 'react';
-import { Modal, ModalVariant } from '@patternfly/react-core';
 import { normalizeClusterVersion } from '../../config';
 import { OcmClusterType } from './types';
 import { AddHostsClusterCreateParams, Cluster, getCluster, handleApiError } from '../../api';
-import RedirectToCluster from '../ui/RedirectToCluster';
 import { getOcmClusterId } from './utils';
 import { usePullSecretFetch } from '../fetching/pullSecret';
 import { ErrorState, LoadingState } from '../ui';
 import { addHostsClusters } from '../../api/addHostsClusters';
+import { AlertsContextProvider } from '../AlertsContextProvider';
+import { POLLING_INTERVAL } from '../../config';
+import AddBareMetalHosts from './AddBareMetalHosts';
 
-const AddBareMetalHostsDialog: React.FC<{
-  isOpen: boolean;
-  closeModal: () => {};
-  cluster: OcmClusterType;
-}> = ({ isOpen, cluster, closeModal }) => {
+type ClusterDetailTabType = {
+  key: number;
+  title: string;
+  contentId: string;
+  id: string;
+  show: boolean;
+  ref: React.RefObject<unknown>;
+};
+
+export const getOCMClusterDetailTab = (
+  tabs: ClusterDetailTabType[],
+  addBareMetalTabRef: React.RefObject<unknown>,
+  displayAddBareMetalHosts: boolean,
+): ClusterDetailTabType => ({
+  key: tabs.reduce((max, tab) => (tab.key > max ? tab.key : max), tabs[0].key) + 1,
+  title: 'Add Hosts',
```

Comment:
```suggestion
  title: 'Bare Metal',
```

I know we want to highlight this new feature so that it gets used (especially until we have something in OCP) but `Add Hosts` is more action-ey than the other tab labels and doesn't quite feel right. Renaming this `Bare Metal` is a little better, but I still find myself thinking this page should eventually just be an action someday (like it would be in OCP).

Reply:
@andybraren , can you please very briefly elaborate what do you mean by turning this feature into an action? This PR is actually about moving it from a kebab-action to a subtab. Maybe we don't want this to be merged ... Or we can rework.

This "tiny" title change gives us option to much wider use case in the future, anyway.

Reply:
Maybe a middle ground can be adding a plus icon to the right of the "Add Hosts"?
It will make this "Tab" look & feel different (like an action):
![image](https://user-images.githubusercontent.com/65043359/98090435-aeb0a900-1e8c-11eb-9056-dece96e966e8.png)

Reply:
I just discovered that the Support tab has a pattern that's very similar to what we're trying to do in this PR. Notice that the tab name is a noun - `Support` - but it has a section for support cases with a button to create a new one:

![image](https://user-images.githubusercontent.com/9122899/98198585-45b24b00-1ef7-11eb-8539-88a35edafd3f.png)

So here's a revised mockup with a tweaked card title (`Bare Metal Discovery` like the [future wizard step](https://marvelapp.com/prototype/7ce7ib3/screen/73593972)) that seems to align us pretty well with OCM's pattern:

![2-ocm-add-hosts](https://user-images.githubusercontent.com/9122899/98200111-e0605900-1efa-11eb-8d3f-3a53ddef64b9.png)

The Actions menu in the top-right corner is disabled in OCM for me, but I'd be interested to see if `Open support case` is an action in there. If it's in there, maybe we should keep "Add Bare Metal Hosts" in there as well and link users to this page/tab when clicked. If not, then maybe we can skip that for now and Nir & I can sync with OCM designers and follow up later.

What do you both think?

Reply:
These kebab actions lead to modal dialogs, as we had the intermediary modal in first version. Unless there is a pattern for it, I would not add this action leading the user to the subtab. Seems to me confusing considering context. In addition, we would need to refactor OCM codebase.

IIUC, the expected change is just the subtab title, right?

Reply:
Sounds good, no action then, just a tab titled `Bare Metal`. Sorry for the long thread to get to that conclusion. 🙂

Reply:
And the card re-titled Bare Metal Discovery*